### PR TITLE
Add SPI to RP2040 

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request adds SPI to the RP2040.

### Testing Strategy

This pull request was tested using a Raspberry Pi Pico.

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: chips/rp2040/src/chip.rs [hunk 1]
```diff
@@ -147,9 +149,14 @@ impl Rp2040DefaultPeripherals<'_> {
...
             adc: adc::Adc::new(),
+            spi0: spi::Spi::new_spi0(),
             sysinfo: sysinfo::SysInfo::new(),
         }
     }
+
+    pub fn set_clocks(&'a self) {
+        self.spi0.set_clocks(&self.clocks);
+    }
 }
```
Comment (bradjc) [inline chips/rp2040/src/chip.rs:160]: Wouldn't it be easier to just pass `clocks` in as an argument to `spi::new()`?

**Review by bradjc — APPROVED** [review]

File: chips/rp2040/src/chip.rs (hunk 1 above)
Comment (alexandruradovici) [reply chips/rp2040/src/chip.rs:160]: I don't think so, it is a circular dependency, as peripherals has  the clock field. I can try though.

Comment (hudson-ayers) [reply chips/rp2040/src/chip.rs:160]: Yeah, that won't work. I used a `setup_circular_dependencies()` method to address that issue for some other chips, which at least made it only require a single call in the case where there were several circular dependencies.

**Review by hudson-ayers — APPROVED** [review]

File: chips/rp2040/src/chip.rs (hunk 1 above)
Comment (alexandruradovici) [reply chips/rp2040/src/chip.rs:160]: I followed your code 😃

**Pull request merged**